### PR TITLE
Enhance point mutation executor

### PR DESCRIPTION
**Pull request by zhang-ivy**
## Description

- Update docstrings and comments
- Fix broken example in docstring 
- Remove `box_dimensions` as input to `PointMutationExecutor`
- Accept `padding` as input to `PointMutationExecutor`
- Accept `box_shape` as input to `PointMutationExecutor`
- Remove `phase` as input to `PointMutationExecutor` and replace with `is_vacuum`
- Remove `solvate` as input to `PointMutationExecutor` and replace with `is_solvated`

## Motivation and context

The above changes 1) improve the readability/usability of the `PointMutationExecutor` and 2) Updates the solvation code to be more practical (in my experience, `box_dimensions` was never used, but I did want to change the padding and box_shape) 

## How has this been tested?

Existing tests: `tests/test_resume.py` and `tests/test_relative_point_mutation_setup.py`

## Change log

```

```

### CODE REVIEW

**Review by mikemhenry — APPROVED** [review]
LGTM

**Review by ijpulidos — COMMENTED** [review]
I have some comments on the new flags for the executor. I feel these could be more telling for users.

File: perses/app/relative_point_mutation_setup.py [hunk 1]
```diff
@@ -77,41 +89,54 @@ class PointMutationExecutor(object):
...
                  allow_undefined_stereo_sdf=False,
+
+                 # Atom mapping parameters
                  extra_sidechain_map=None,
                  demap_CBs=False,
-                 solvate=True,
+
+                 # Solvation parameters
+                 is_vacuum=False,
+                 is_solvated=False,
```
[the remark this answers is not in the document]
Comment (ijpulidos) [reply perses/app/relative_point_mutation_setup.py:124]: There might be some logic needed to check if it makes sense to solvate the system or not, but I think this should also happen inside `_solvate()` method Such that it warns (or errors out) if users are trying to solvate an "unsolvatable" system. Something similar should happen when charge changes occur, these should be dealt in the `_handle_charge_changes()` method.

**Comment by zhang-ivy** [comment]
@ijpulidos:

>I'm struggling a bit in understanding why we need two parameters for this. Shouldn't it be that if it's vacuum then it isn't solvated and if it is solvated then it isn't vacuum? Sounds like this could be reduced to a single parameter, unless it makes sense to have some topology that isn't vacuum but it's also not solvated.

 I spent a lot of time thinking about the flags I ended up choosing, so let me explain them here. 

The are actually many use cases, let me outline them here:
- If the user supplies un-solvated input files (`is_solvated=False`)
    - If the user desires to run in vacuum phase (`is_vacuum=True`)
        - skip counterion, skip solvation
    - If the user desires to run in solvent/complex phase (`is_vacuum=False`)
        - add counterion, solvate
- If the user supplies solvated input files (`is_solvated=True`)
    - assert that `is_vacuum=False` and then add counterion, but skip solvation
    
This could probably be designed better in the future with the new API, but I think we should keep these two flags as is for now. 

> I think it is more intuitive and readable to think about if we want to solvate the tolopology

Yes, I originally thought so too, which is why I named the flag `solvate` before, but I wanted to group these two cases together:

```
    - If the user desires to run in vacuum phase (`is_vacuum=True`)
        - skip counterion, skip solvation
    - If the user desires to run in solvent/complex phase (`is_vacuum=False`)
        - add counterion, solvate
```
if I name the flag `solvate`, then they won't be grouped together. I think the logic of the code flows better this way. 

>BUT we would also need another flag if we want to handle the charges, say handle_charges=True/False, instead of relying on if it's solvated and not vacuum or such combinations. 

I introduced a flag to allow the user to control whether we add a counterion in https://github.com/choderalab/perses/pull/1030, but I think that PR will require further testing, so I suggest we move it to the next milestone/release. 
Note that this flag allows the user to control the addition of the counterion, but only if the phase is not vacuum. If the phase is vacuum, we should never add a counterion or solvate, which is why the `is_vacuum` flag exists.

>There might be some logic needed to check if it makes sense to solvate the system or not, but I think this should also happen inside _solvate() method Such that it warns (or errors out) if users are trying to solvate an "unsolvatable" system. Something similar should happen when charge changes occur, these should be dealt in the _handle_charge_changes() method.

Sure, could you create an issue for this, I don't think its necessary for this release, but it would be nice to have in the future.

**Comment by ijpulidos** [comment]
@zhang-ivy Thanks for clarifying this to me. I think this is okay for now, but we might need to think about this in future releases, I think we already see where things can be improved. My suggestions were never blocking, by the way, so LGTM!

**Comment by mikemhenry** [comment]
@ijpulidos can you make some issue(s) out of https://github.com/choderalab/perses/pull/1024/#issuecomment-1148875313 
@zhang-ivy can you make sure I did this merge https://github.com/choderalab/perses/pull/1024/commits/ce6016d562d46e6b37a9bfae2db726b27644e2c0  correctly?

**Comment by zhang-ivy** [comment]
@mikemhenry : Just reviewed the merge -- it looks good!

**Comment by zhang-ivy** [comment]
@mikemhenry : This should be good to merge now, I think?

**Pull request merged**